sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,138 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import {
+  createFakeInstitutionLocation,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  Institution,
+  InstitutionLocation,
+  Student,
+} from "@sims/sims-db";
+import {
+  createTestingAppModule,
+  BEARER_AUTH_TYPE,
+  AESTGroups,
+  getAESTToken,
+} from "../../testHelpers";
+
+describe("StudentMinistryController(e2e)-searchStudents", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let studentRepo: Repository<Student>;
+  let applicationRepo: Repository<Application>;
+  const endpoint = "/aest/student/search";
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    studentRepo = dataSource.getRepository(Student);
+    applicationRepo = dataSource.getRepository(Application);
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+  });
+
+  it("Should find the student by part of first name.", async () => {
+    // Arrange
+    const { student } = await saveStudentWithApplicationForCollegeF(
```

Comment:
No need to save a Student associated with an application. Ministry has full access to any student.
Please remove the application associations, a saveFakeStudent should be enough for this E2E file.

Reply:
updated